clippy_lints/src/casts/cast_slice_different_sizes.rs

```diff
@@ -59,7 +59,7 @@ pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, msrv: &Option<RustcVe
                         from_slice_ty, from_size, to_slice_ty, to_size,
                     ),
                     |diag| {
-                        let cast_expr = match expr.kind {
+                        let cast_expr = match expr.peel_blocks().kind {
                             ExprKind::Cast(cast_expr, ..) => cast_expr,
                             _ => unreachable!("expr should be a cast as checked by expr_cast_chain_tys"),
                         };
```

Comment:
Thanks!

I'd suggest returning `cast_expr` (or its span) from `expr_cast_chain_tys` so that you don't need to keep these two places in sync

Reply:
This ends up simplifying things and actually led me to notice a bug in the output too (which is fixed in the latest commit), thanks.